### PR TITLE
search plugin: do not crash on missing lang options, inform instead

### PR DESCRIPTION
**Pull request by ultrabug**
- fallback to english on missing search.lang option
- support search.lang territory pt_BR -> pt

### CODE REVIEW

File: mkdocs/contrib/search/__init__.py [hunk 1]
```diff
@@ -14,19 +14,29 @@ class LangOption(config_options.OptionallyRequired):
...
         if isinstance(value, str):
             value = [value]
         elif not isinstance(value, (list, tuple)):
             raise config_options.ValidationError('Expected a list of language codes.')
-        for lang in value:
-            if lang != 'en' and not self.lang_file_exists(lang):
-                raise config_options.ValidationError(
-                    f'"{lang}" is not a supported language code.'
-                )
+        for lang in list(value):
```
Comment (oprypin) [inline mkdocs/contrib/search/__init__.py:27]: What's the reason to call `list(value)` here? From basic understanding, there is no difference at all vs not calling it.

File: mkdocs/contrib/search/__init__.py [hunk 2]
```diff
@@ -14,19 +14,29 @@ class LangOption(config_options.OptionallyRequired):
     """ Validate Language(s) provided in config are known languages. """
 
     def lang_file_exists(self, lang):
```
Comment (oprypin) [inline mkdocs/contrib/search/__init__.py]: This function is no longer about existence, should rename it

File: mkdocs/contrib/search/__init__.py [hunk 3]
```diff
@@ -14,19 +14,29 @@ class LangOption(config_options.OptionallyRequired):
...
-        path = os.path.join(base_path, 'lunr-language', f'lunr.{lang}.js')
-        return os.path.isfile(path)
+        for lang_part in lang.split("_"):
+            lang_part = lang_part.lower()
+            if os.path.isfile(os.path.join(base_path, 'lunr-language', f'lunr.{lang_part}.js')):
+                return lang_part
+        return False
```
Comment (oprypin) [inline mkdocs/contrib/search/__init__.py]: Perhaps return None now

**Comment by ultrabug** [comment]
Good for me now :+1:

File: mkdocs/tests/search_tests.py [hunk 4]
```diff
@@ -60,11 +60,18 @@ def test_lang_bad_type(self):
...
+        value = option.validate(['foo'])
+        self.assertEqual(['en'], value)
 
     def test_lang_good_and_bad_code(self):
         option = search.LangOption()
-        self.assertRaises(ValidationError, option.validate, ['en', 'foo'])
+        value = option.validate(['en', 'foo'])
+        self.assertEqual(['en'], value)
+
+    def test_lang_good_and_bad_and_territory(self):
```
Comment (oprypin) [inline mkdocs/tests/search_tests.py]: Uhhh let's not accidentally imply that some particular territory is bad :scream: 

What would the word be... "missing"?

**Comment by ultrabug** [comment]
Thanks for the review @oprypin

**Comment by oprypin** [comment]
Ah thanks for the great fix.

@ultrabug so you probably saw that I mentioned that a release is coming on Sunday [today]
but it seems like there's still some language PRs that we could merge now. So let's not rush with the release.

**Comment by ultrabug** [comment]
Yes mate indeed, I'll check the pt_BR PR out later today, thanks

**Pull request merged**